src/api/endpoints/societies.py

```diff
@@ -50,19 +52,32 @@ def get(self, society_id=None):
         """Get Society(ies) details."""
         if society_id:
             society = Society.query.get(society_id)
-            return find_item(society)
-
+        elif request.args.get('name'):
+            society = Society.query.filter_by(
+                name=request.args.get('name')).first()
         else:
-            search_term = request.args.get('q')
-
-            if search_term:
-                society = Society.query.filter_by(name=search_term).first()
-                return find_item(society)
-
             # if no search term has been passed, return all societies in DB
             societies = Society.query
             return paginate_items(societies)
 
+        if society:
+            society_logged_activities = LoggedActivity.query.filter_by(
+                society_id=society.uuid).all()
+
+            data, _ = society_schema.dump(society)
```

Comment:
Why just a dash to signify errors?
Wouldn't an actual variable name be more useful?

Reply:
I use the _ because that value is a don't care at that point. If all the logic above holds true, there is no possibility they will be errors.